src/google/protobuf/compiler/objectivec/objectivec_file.cc

```diff
@@ -402,21 +402,46 @@ void FileGenerator::GenerateSource(io::Printer *printer) {
   //  deprecated-declarations suppression is only needed if some place in this
   //    proto file is something deprecated or if it references something from
   //    another file that is deprecated.
+  //  dollar-in-identifier-extension is needed because we use references to
+  //    objc class names that have $ in identifiers.
   printer->Print(
       "// @@protoc_insertion_point(imports)\n"
       "\n"
       "#pragma clang diagnostic push\n"
-      "#pragma clang diagnostic ignored \"-Wdeprecated-declarations\"\n");
+      "#pragma clang diagnostic ignored \"-Wdeprecated-declarations\"\n"
+      "#pragma clang diagnostic ignored \"-Wdollar-in-identifier-extension\"\n");
```

Comment:
move this under a `if (fwd_decls.size() > 0) {` so it is only added when needed (like the direct-ivar-access one)

Reply:
Done